### PR TITLE
Feature lpdev 807 add settlement statuses cqrs

### PR DESCRIPTION
**Pull request by LatheLykke**

### CODE REVIEW

File: src/Lykke.Service.PayInternal/Modules/CqrsModule.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        public const string InternalBoundedContext = "lykkepay-internal";
+
+        public CqrsModule(IReloadingManager<AppSettings> appSettings)
+        {
+            _appSettings = appSettings;
+        }
+
+        protected override void Load(ContainerBuilder builder)
+        {
+            builder.RegisterType<SilentChaosKitty>()
```
Comment (tarurar) [inline src/Lykke.Service.PayInternal/Modules/CqrsModule.cs]: I'd use regular ChaosKitty here.

File: src/Lykke.Service.PayInternal.Core/Domain/PaymentRequests/PaymentRequestStatus.cs [hunk 2]
```diff
@@ -43,6 +43,21 @@ public enum PaymentRequestStatus
         /// <summary>
         /// An error occurred during processing payment request.
         /// </summary>
-        Error
+        Error,
+
+        /// <summary>
+        /// Payment request is accepted to settlement.
+        /// </summary>
+        SettlementInProgress,
```
Comment (tarurar) [inline src/Lykke.Service.PayInternal.Core/Domain/PaymentRequests/PaymentRequestStatus.cs]: Are these new status being mapped to Invoice statuses?

File: src/Lykke.Service.PayInternal.Contract/PaymentRequest/PaymentRequestPublicStatuses.cs [hunk 3]
```diff
@@ -44,5 +44,20 @@ public static class PaymentRequestPublicStatuses
         /// Refund error
         /// </summary>
         public const string RefundError = "REFUND_ERROR";
+
+        /// <summary>
+        /// Payment request is accepted to settlement.
+        /// </summary>
+        public const string SettlementInProgress = "SETTLEMENT_INPROGRESS";
```
Comment (tarurar) [inline src/Lykke.Service.PayInternal.Contract/PaymentRequest/PaymentRequestPublicStatuses.cs]: These new public statuses must be mapped to public statuses in public API: https://github.com/LykkeCity/LykkePayAPI/blob/dev/src/Lykke.Service.PayAPI/ConvertExtensions.cs#L60

File: src/Lykke.Service.PayInternal.Core/Domain/PaymentRequests/PaymentRequest.cs [hunk 4]
```diff
@@ -59,5 +59,7 @@ public bool StatusValidForSettlement()
         }
 
         public string Initiator { get; set; }
+
+        public string SettlementErrorDescription { get; set; }
```
Comment (tarurar) [inline src/Lykke.Service.PayInternal.Core/Domain/PaymentRequests/PaymentRequest.cs]: I believe ProcessingError property is intended to be used for this purpose. Please consider the possibility of using it.

File: src/Lykke.Service.PayInternal/Modules/CqrsModule.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                        settlementTransport,
+                        _appSettings.CurrentValue.PaySettlementCqrs.SerializationFormat,
+                        environment: _appSettings.CurrentValue.PaySettlementCqrs.Environment)),
+
+                    Register.BoundedContext(InternalBoundedContext)
+                        .ListeningEvents(typeof(SettlementTransferToMarketQueuedEvent),
+                            typeof(SettlementTransferredToMerchantEvent), typeof(SettlementErrorEvent))
+                        .From(_appSettings.CurrentValue.PaySettlementCqrs.SettlementBoundedContext)
+                        .On(_appSettings.CurrentValue.PaySettlementCqrs.EventsRoute)
+                        .WithProjection(ctx.Resolve<SettlementProjection>(),
```
Comment (tarurar) [inline src/Lykke.Service.PayInternal/Modules/CqrsModule.cs]: If I am not mistaken the commands should be used when there is a state change.

File: src/Lykke.Service.PayInternal.Core/Domain/PaymentRequests/PaymentRequestStatus.cs (hunk 2 above)
Comment (LatheLykke) [reply src/Lykke.Service.PayInternal.Core/Domain/PaymentRequests/PaymentRequestStatus.cs]: LPDEV-900 and LPDEV-901 are created.

File: src/Lykke.Service.PayInternal.Contract/PaymentRequest/PaymentRequestPublicStatuses.cs (hunk 3 above)
Comment (LatheLykke) [reply src/Lykke.Service.PayInternal.Contract/PaymentRequest/PaymentRequestPublicStatuses.cs]: LPDEV-902 is created

File: src/Lykke.Service.PayInternal/Modules/CqrsModule.cs (hunk 5 above)
Comment (LatheLykke) [reply src/Lykke.Service.PayInternal/Modules/CqrsModule.cs]: There is no CQRS workflow in this project. Using saga with commands leads to surplus complexity in this case. I think will be better to add this feature to sage on moving this project to CQRS.

**Pull request merged**